src/main/java/io/fabric8/jenkins/openshiftsync/JenkinsUtils.java

```diff
@@ -575,6 +575,13 @@ public static WorkflowJob getJobFromBuild(Build build) {
 		if (StringUtils.isEmpty(buildConfigName)) {
 			return null;
 		}
+
+		WorkflowJob job = BuildConfigToJobMap.getJobFromBuildConfigNameNamespace(buildConfigName, 
+		        build.getMetadata().getNamespace());
```

Comment:
i guess this is why?  we don't have the uid at this point in the code so you switched the cache to use a key that's more universally available?